### PR TITLE
Update seastar submodule

### PR DESCRIPTION
**Pull request by tchaikov**
* seastar e0d515b6...70349b74 (33):
  > util/log: drop unused function
  > util/log, rpc, core: use compile-time formatting with fmtlib >= 8.0
  > Fix edge case in memory sampler at OOM
  > exp/geo distribution benchmark
  > Additional allocation tests
  > Remove null pointer check on free hot path
  > Optimize final part of allocation hot path
  > Optimize zero size checking in allocator
  > memory: Optimize free fast path
  > memory: Optimize small alloc alloation path
  > memory: Limit alloc_sites size
  > memory: Add general comment about sampling strategy
  > memory: Use probabilistic sampler
  > util: Adapt memory sampler to seastar
  > util: Import Android Memory Sampler
  > memory: Use separate small pool for tracking sampled allocations
  > memory: Support enabling memory profiling at runtime
  > util/source_location-compat: mark `source_location::current()` consteval
  > build: use new behavior defined by CMP0155 when building C++ modules
  > circleci: build with C++20 modules enabled
  > seastar.cc: replace cryptopp with gnutls when building seastar modules
  > alien: include used header
  > seastar.cc: include used headers in the global purview
  > docker: install clang-tools-17
  > net/tcp: generate a random src_port hashed to current shard if smp::count > 1
  > net, websocket: replace Crypto++ calls with GnuTLS
  > README-DPDK.md: point user to DPDK's quick start guide
  > reactor: print fatal error using logger as well
  > Avoid ping-pong in spinlock::lock
  > memory: Add allocator perf tests
  > memory: Add a basic sized deletion test
  > Prometheus: Disable Prometheus protobuf with a configuration
  > treewide: bring back prometheus protobuf support
* test/manual/sstable_scan_footprint_test: update to adapt to the
  breaking change of "memory: Use probabilistic sampler" in seastar

### CODE REVIEW

File: cql3/statements/batch_statement.cc [hunk 1]
```diff
@@ -215,10 +215,10 @@ void batch_statement::verify_batch_size(query_processor& qp, const std::vector<m
                     mutations.size(), fmt::join(ks_cf_pairs, ", "), size, type, threshold, size - threshold);
         };
         if (size > fail_threshold) {
-            _logger.error(error("FAIL", fail_threshold).c_str());
+            _logger.error("{}", error("FAIL", fail_threshold).c_str());
```
Comment (avikivity) [inline cql3/statements/batch_statement.cc:218]: Please push separately, I like submodule updates to be in the top-level.

File: tools/scylla-sstable.cc [hunk 2]
```diff
@@ -1752,7 +1752,7 @@ class json_mutation_stream_parser {
 #else
             auto parse_error = fmt::format(msg, std::forward<decltype(args)>(args)...);
 #endif
```
Comment (avikivity) [inline tools/scylla-sstable.cc:1754]: cqlsh update looks unrelated

File: cql3/statements/batch_statement.cc (hunk 1 above)
Comment (tchaikov) [reply cql3/statements/batch_statement.cc:218]: sure. will extract the treewide one out.

File: tools/scylla-sstable.cc (hunk 2 above)
Comment (tchaikov) [reply tools/scylla-sstable.cc:1754]: dropped.

File: cql3/statements/batch_statement.cc (hunk 1 above)
Comment (tchaikov) [reply cql3/statements/batch_statement.cc:218]: extracted into #16612. i will drop it from this PR once the build passes.

File: test/manual/sstable_scan_footprint_test.cc [hunk 3]
```diff
@@ -266,15 +266,15 @@ void test_main_thread(cql_test_env& env) {
     stats_collector sc(sem, stats_collector_params);
     try {
         auto _ = sc.collect();
-        memory::set_heap_profiling_enabled(true);
+        memory::set_heap_profiling_sampling_rate(100);
```
Comment (tchaikov) [inline test/manual/sstable_scan_footprint_test.cc:269]: @michoecho @avikivity  a breaking change sneaked in anyway. shall we add a follow-up change to bring this API back, like implemented with a reasonable default sampling rate ?

**Comment by tchaikov** [comment]
the commit of [test: randomized_nemesis_test: move std::variant formatter up](https://github.com/scylladb/scylladb/pull/16610/commits/ea334f039e5baa5f450cff03776f3a5a8228434a) is extracted into #16616

**Comment by tchaikov** [comment]
> #### 🔴 CI State: FAILURE
> 
> ✅ - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/5522/) ✅ - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/5522/) ❌ - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/6934/)
> #### Failed Tests (1/23683):
> 
>     * [test_cql_query_filtering_without_indexes](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/5522/testReport/junit/cql_tests/TestTruncate/Tests___dtest___test_cql_query_filtering_without_indexes) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_cql_query_filtering_without_indexes)
> 
> 
> #### Build Details:
> 
>     * Duration: 1 hr 53 min
> 
>     * Builder: spider3.cloudius-systems.com

new flaky test, tracked by #16621

**Comment by tchaikov** [comment]
> #### 🔴 CI State: FAILURE
> ❌ - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/5541/)
> 
> #### Build Details:
> * Duration: 10 min
> * Builder: spider3.cloudius-systems.com

Clang frontend crashed.

File: test/manual/sstable_scan_footprint_test.cc (hunk 3 above)
Comment (avikivity) [reply test/manual/sstable_scan_footprint_test.cc:269]: I think we can let it slip, if it hits anyone else (unlikely), it's a not in any important place.

Comment (tchaikov) [reply test/manual/sstable_scan_footprint_test.cc:269]: ack. thanks.
